tests/rptest/tests/data_transforms_test.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+
+
+class WasmDeployException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return repr(self.message)
+
+
+class DataTransformsTest(RedpandaTest):
+    """
+    Tests related to WebAssembly powered data transforms
+    """
+    topics = [TopicSpec(partition_count=9), TopicSpec(partition_count=9)]
+
+    def __init__(self, test_context):
+        super(DataTransformsTest, self).__init__(test_context=test_context)
+        self._ctx = test_context
+        self._rpk = RpkTool(self.redpanda)
+
+    def _enable_wasm(self):
+        """
+        Enable data transforms and restart the cluster.
+        """
+        self._rpk.cluster_config_set("data_transforms_enabled", True)
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+
+    def _deploy_wasm(self, name: str):
+        """
+        Deploy a wasm transform and wait for all processors to be running.
+        """
+        [input_topic, output_topic] = self.topics
+
+        def do_deploy():
+            self._rpk.deploy_wasm(name, input_topic.name, output_topic.name)
+            return True
+
+        wait_until(
+            do_deploy,
+            timeout_sec=30,
+            backoff_sec=5,
+            err_msg=f"unable to deploy wasm transform {name}",
+            retry_on_exc=True,
+        )
```

Comment:
Always `return True` seems fishy. Is there a possibility that deploying wasm fails without throwing an exception?

Reply:
Not at the moment. The RPK command returns `None` and will throw if the exit code is not `0` (which means the deploy failed).